### ISSUE TITLE
Fixing file upload for Android in react-native-stream-chat

### ISSUE THREAD

**Issue by joenas** [GetStream/stream-chat-js#1215]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `stream-chat@8.14.1` for the project I'm working on.

When using https://github.com/GetStream/stream-chat-react-native, I noticed that it didn't work to upload any sort of files on Android (photos from library, camera or files). 

This is the error I got

```
Failed to send url request: https://chat.stream-io-api.com/channels/messaging/CHANNEL_ID
java.lang.IllegalArgumentException: multipart != application/x-www-form-urlencoded
	at okhttp3.MultipartBody$Builder.setType(MultipartBody.kt:241)
	at com.facebook.react.modules.network.NetworkingModule.constructMultipartBody(NetworkingModule.java:688)
	at com.facebook.react.modules.network.NetworkingModule.sendRequestInternal(NetworkingModule.java:442)
	at com.facebook.react.modules.network.NetworkingModule.sendRequest(NetworkingModule.java:236)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
	at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:188)
	at com.facebook.jni.NativeRunnable.run(Native Method)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27)
	at android.os.Looper.loop(Looper.java:223)
	at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:228)
	at java.lang.Thread.run(Thread.java:923)
I  [AxiosError: Network Error]
```

iOS works fine so I imagine it's something with `okhttp3` or React-native itself. Some other issues like https://github.com/facebook/react-native/issues/16149 & https://github.com/GetStream/stream-chat-react-native/issues/1640 mentions this. 

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/stream-chat/dist/browser.js b/node_modules/stream-chat/dist/browser.js
index c4ecff8..4736968 100644
--- a/node_modules/stream-chat/dist/browser.js
+++ b/node_modules/stream-chat/dist/browser.js
@@ -7669,7 +7669,7 @@ var StreamChat = /*#__PURE__*/function () {
       var data = addFileToFormData(uri, name, contentType);
       if (user != null) data.append('user', JSON.stringify(user));
       return this.doAxiosRequest('post', url, data, {
-        headers: data.getHeaders ? data.getHeaders() : {},
+        headers: data.getHeaders ? data.getHeaders() : { 'Content-Type': 'multipart/form-data' },
         // node vs browser
         config: {
           timeout: 0,
```

Another possible solution is adding `getHeaders()` to `data` in `addFileToFormData` but this seems more straight forward. From what I've gathered, `getHeaders()` is a Node-thing, hence not present in my case.

Incidentally, this fix will generate a warning in [getResizedImageUrl](https://github.com/GetStream/stream-chat-react-native/blob/develop/package/src/utils/getResizedImageUrl.ts#L24) that I haven't seen before but alas. 

Hopefully there's a way to fix this permanently! Cheers

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by santhoshvai** [comment]
Hi @joenas 

the change you mentioned is already available in stream-chat@8.14.4 which has this [PR](https://github.com/GetStream/stream-chat-js/pull/1202)

and this version was first used in stream-chat-react-native@5.22.1

Please upgrade your stream-chat-react-native version and that should solve your issue

**Comment by joenas** [comment]
Great @santhoshvai ! Thanks 🎉  When I looked I figured I was on the latest version but apparently not 🙂
